tokio/src/sync/notify.rs

```diff
@@ -16,7 +16,7 @@ use std::marker::PhantomPinned;
 use std::panic::{RefUnwindSafe, UnwindSafe};
 use std::pin::Pin;
 use std::ptr::NonNull;
-use std::sync::atomic::Ordering::{self, Acquire, Relaxed, Release, SeqCst};
+use std::sync::atomic::Ordering::{self, Acquire, Relaxed, Release, AcqRel, SeqCst};
```

Comment:
Wait, does this PR not eliminate all usages of SeqCst?

Reply:
> Wait, does this PR not eliminate all usages of SeqCst?

Yes, indeed, other parts of this model also use SeqCst, and I believe it's feasible to relax these orderings to Acquire/Release. However, I have not made any modifications to the SeqCst usage in other areas, as these usages do not trigger alerts in the detector(possibly because the call chain does not involve related function calls). If possible, I would like to initially ease it to Acquire/Release and then submit a new pull request.

Reply:
I don't know how your detector works, but mixing SeqCst and non-SeqCst orderings is never the right answer. Newer.

Looking over the code, I notice that there are atomics in both the `Notify`, but also the waiters. Before I can accept a PR relaxing any of them, I'm going to have to review the code for whether it currently relies on the SeqCst guarantees to synchronize between those different atomics.

Reply:
> I don't know how your detector works, but mixing SeqCst and non-SeqCst orderings is never the right answer. Never.
> 
> Looking over the code, I notice that there are atomics in both the `Notify`, but also the waiters. Before I can accept a PR relaxing any of them, I'm going to have to review the code for whether it currently relies on the SeqCst guarantees to synchronize between those different atomics.


What you've mentioned is quite insightful. I've noticed that there are numerous instances of ordering being used improperly within the community. Therefore, my aim is to develop an automated system to detect such issues. Currently, this project is still experimental and undergoing continuous optimization. Regarding the issue you've highlighted, I believe I've pinpointed the root cause. It seems that during the traversal of the call graph, the regular expression I crafted failed to match the relevant atomic operations. This is an aspect I am actively working to resolve. It's important to clarify that this isn't an issue with the detection pattern per se, but rather a specific challenge I'm addressing💗.